src/EditorFeatures/CSharpTest/UseLocalFunction/UseLocalFunctionTests.cs

```diff
@@ -1191,5 +1191,295 @@ void M()
+@"using System;
+
+class Program
+{
+    static void Main(string[] args)
+    {
+        Func<object, string> [||]f = x => """";
+        M(f);
+    }
+
+    public static void M(Func<string, object> expected) { }
+}",
+@"using System;
+
+class Program
+{
+    static void Main(string[] args)
+    {
+        string f(object x) => """";
+        M(f);
+    }
+
+    public static void M(Func<string, object> expected) { }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseLocalFunction)]
+        [WorkItem(23118, "https://github.com/dotnet/roslyn/issues/23118")]
+        public async Task TestAvailableIfConvertedToCoContraVariantDelegate2()
+        {
+            await TestInRegularAndScript1Async(
+@"using System;
+
+class Program
+{
+    static void Main(string[] args)
+    {
+        Func<string, string> [||]f = x => """";
+        M(f);
+    }
+
+    public static void M(Func<string, object> expected) { }
+}",
+@"using System;
+
+class Program
+{
+    static void Main(string[] args)
+    {
+        string f(string x) => """";
+        M(f);
+    }
+
+    public static void M(Func<string, object> expected) { }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseLocalFunction)]
+        [WorkItem(23118, "https://github.com/dotnet/roslyn/issues/23118")]
+        public async Task TestAvailableIfConvertedToCoContraVariantDelegate3()
+        {
+            await TestInRegularAndScript1Async(
+@"using System;
+
+class Program
+{
+    static void Main(string[] args)
+    {
+        Func<object, object> [||]f = x => """";
+        M(f);
+    }
+
+    public static void M(Func<string, object> expected) { }
+}",
+@"using System;
+
+class Program
+{
+    static void Main(string[] args)
+    {
+        object f(object x) => """";
+        M(f);
+    }
+
+    public static void M(Func<string, object> expected) { }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseLocalFunction)]
+        [WorkItem(22672, "https://github.com/dotnet/roslyn/issues/22672")]
+        public async Task TestMissingIfAdded()
+        {
+            await TestMissingAsync(
+@"using System;
+using System.Linq.Expressions;
+
+class Enclosing<T> where T : class
+{
+  delegate T MyDelegate(T t = null);
+
+  public class Class
+  {
+    public void Caller()
+    {
+      MyDelegate [||]local = x => x;
+
+      var doubleDelegate = local + local;
+    }
+  }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseLocalFunction)]
+        [WorkItem(22672, "https://github.com/dotnet/roslyn/issues/22672")]
+        public async Task TestMissingIfUsedInMemberAccess()
+        {
+            await TestMissingAsync(
+@"using System;
+
+class Enclosing<T> where T : class
+{
+  delegate T MyDelegate(T t = null);
+
+  public class Class
+  {
+    public void Caller()
+    {
+      MyDelegate [||]local = x => x;
+
+      local.Invoke();
```

Comment:
:bulb: Would be good to file a follow-up issue (feature request) to handle the case where the only member access is `Invoke`.